### DISCUSSION TITLE
Why does subpixel averaging occur so many times in my setup?

### DISCUSSION THREAD

**Discussion by cyasing** [NanoComp/meep#2320] [Q&A] [answered]
My setup size is 27x27, 2D with a block geometry of 8x10.5, with a unique material_function medium for it. There are 4 different line sources. I was just wondering why subpixel-averaging occurs so many times, as can be seen here:

![image](https://user-images.githubusercontent.com/59028917/203618209-d09c491a-81e2-495a-8824-66c562a43a6c.png)

**Comment by smartalecH** [comment, marked as answer]
Subpixel smoothing only "occurs" once. You see multiple updates depending on how many "chunks" are in your simulation, and how long it takes for each chunk. 

If you notice the smoothing is particularly slow, then you can speed it up with almost perfect parallel efficiency using MPI (in contrast, the FDTD algorithm itself will not experience this type of scaling due to communication costs). There are caveats with material functions, however, where this scaling breaks down due to the python GIL (although subpixel smoothing is disabled by default for material functions due to the expensive brute force quadrature required).

**Reply by cyasing** [reply]
I see.
